services/community-site/src/utils/validators.ts

```diff
@@ -0,0 +1,34 @@
+import { Validator, ValidatorWithStats, YieldedValidator } from '../interfaces/Validator';
+
+const calulcateValidatorYield = (validators: Validator[]): ValidatorWithStats[] | Validator[] => {
+  const hasStats = (validators[0] as ValidatorWithStats)?.pbftsProduced !== 0;
+  if (!hasStats) {
+    return validators;
+  }
+  const validatorsWithStats = validators as ValidatorWithStats[];
+  const validatorsWithStake: YieldedValidator[] = validatorsWithStats.map((v) => {
```

Comment:
I don't see the point of this `YieldedValidator` interface. The name doesn't even make sense because you added the "blocksPerStake" field, not the "yield"